Update Avatar's Spec.md and Migration.md (#21702)
microsoft/fluentui: packages/react-avatar/SPEC.md

The Avatar component represents a person or entity. It displays the person's image, initials, or an icon, and can be either circular or square.

> Note: The Avatar control has been mostly implemented already. Visit [Avatar Storybook Examples](https://fluentuipr.z22.web.core.windows.net/pull/21679/react-components/storybook/index.html?path=/docs/components-avatar--default) to see the current state of the implementation.

Reviewer: Should we use `heads/master` instead of a specific PR?
```suggestion
Note: The Avatar control has been mostly implemented already. Visit [Avatar Storybook Examples](https://fluentuipr.z22.web.core.windows.net/heads/master/react-components/storybook/index.html?path=/docs/components-avatar--default) to see the current state of the implementation.
```

Author: Good catch, thanks. I meant to have the link point to master, and pasted the wrong one.

## Sample Code

Display a user's initials:

```jsx
<Avatar name="Miguel Garcia" />
```

> Display an image of a user:

Reviewer 2: Nit: we should be consistent and reword this to "Display a user's image" or "Display user image"

```jsx
<Avatar size={72} name="Mona Kane" image={{ src: './MonaKane.jpg' }} />
```

Display an icon only:

```jsx
<Avatar aria-label="Team" icon={<PeopleTeamRegular />} shape="square" />
```

> Displaying a badge:

Reviewer: nit: To be consistent above this should read `Display` instead of `Displaying`
```suggestion
Display a badge:
```

   * accessibility tools.
   */
  name?: string;

  /**
   * Size of the avatar in pixels.
   *
   * Size is restricted to a limited set of supported values recommended for most uses (see `AvatarSizeValue`) and
   * based on design guidelines for the Avatar control.
   *
   * If a non-supported size is neeeded, set `size` to the next-smaller supported size, and set `width` and `height`
   * to override the rendered size.
   *
   * For example, to set the avatar to 45px in size:
   * `<Avatar size={40} style={{ width: '45px', height: '45px' }} />`
   *
   * @defaultvalue 32
   */
  size?: 20 | 24 | 28 | 32 | 36 | 40 | 48 | 56 | 64 | 72 | 96 | 120 | 128;

  /**
   * The avatar can have a circular or square shape.
   * @defaultvalue circular
   */
  shape?: 'circular' | 'square';

  /**
   * Optional activity indicator
   * * active: the avatar will be decorated according to activeAppearance
   * * inactive: the avatar will be reduced in size and partially transparent
   * * unset: normal display
   *
   * @defaultvalue unset
   */
  active?: 'active' | 'inactive' | 'unset';

  /**
   * The appearance when `active="active"`
   *
   * @defaultvalue ring
   */

>   activeAppearance?: 'ring' | 'shadow' | 'ring-shadow';

Reviewer 2: I noticed that we tend to use camel case in string unions rather than hyphens. Should "ring-shadow" be "ringShadow" here?

Author: Yeah that's a good point; we have some inconsistency in the library about camelCase vs. snake-case for attribute values. (Tooltip and Menu's `positioning` attributes use snake-case like `positioning="above-start"` as well). Given that we're already in RC, it might be too late to fix this up.

Reviewer 2: > Given that we're already in RC, it might be too late to fix this up.

Agreed. This is probably a good topic to discuss in a tech sync or RFC so we can have greater consistency going forward.

The Avatar is non-interactive.

- **Keyboard** - Not keyboard focusable.
- **Mouse** - Nothing
- **Touch** - Nothing

## Accessibility

The Avatar presents as a single img element to accessibility tools, regardless of what it is displaying (image, initials, or icon).

- The Avatar's root has `role="img"`
- All other slots have `aria-hidden="true"`.
- The `<img>` additionally has `alt="" role="presentation"`

> The Avatar root's label will be determined using the following priority:

Reviewer: nit: Use present tense
```suggestion
The Avatar root's label is determined using the following priority:
```
